ext/intl/timezone/timezone_methods.cpp

```diff
@@ -179,9 +176,8 @@ U_CFUNC PHP_FUNCTION(intltz_create_enumeration)
 		/* else call string version */
 		se = TimeZone::createEnumeration(Z_STRVAL_P(arg));
 	} else {
-		intl_error_set(NULL, U_ILLEGAL_ARGUMENT_ERROR,
-			"invalid argument type", 0);
-		RETURN_FALSE;
+		zend_argument_type_error(1, "invalid argument type");
+		RETURN_THROWS();
```

Comment:
Why not use `Z_PARAM_STR_OR_LONG` and if you get a string check if it is numeric?

For numeric floats possibly emit the deprecation warning with `zend_incompatible_double_to_long_error()`